### PR TITLE
ROCm dlpack support

### PR DESCRIPTION
**Pull request by rahulbatra85**
Addresses test failure related to DLPack support in array_interoperability_tests.py

### CODE REVIEW

File: tests/array_interoperability_test.py [hunk 1]
```diff
@@ -215,14 +212,13 @@ def testNumpyToJax(self, shape, dtype):
      for shape in all_shapes
      for dtype in torch_dtypes))
   @unittest.skipIf(numpy_version < (1, 22, 0), "Requires numpy 1.22 or newer")
-  @jtu.skip_on_devices("gpu")
```
Comment (hawkinsp) [inline tests/array_interoperability_test.py:218]: Revert this change.

(NumPy simply doesn't accept GPU dlpacks from any vendor... maybe let's add a comment to that effect?)

File: jax/_src/dlpack.py [hunk 2]
```diff
@@ -57,6 +57,14 @@ def from_dlpack(dlpack):
     gpu_backend = xla_bridge.get_backend("cuda")
   except RuntimeError:
     gpu_backend = None
+
+  #Try ROCm if CUDA backend not found
```
Comment (hawkinsp) [inline jax/_src/dlpack.py:61]: Nit: Space before "Try"

**Review by hawkinsp — COMMENTED** [review]
Is this change sufficient by itself to get things working? Don't we also need a TF change?

If nothing else things like this must have the wrong vendor ID with this code:
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/xla/python/dlpack.cc;drc=6f62beb16126e0840dac5b405ab6284d65b716a2;l=227

File: jax/_src/dlpack.py (hunk 2 above)
Comment (rahulbatra85) [reply jax/_src/dlpack.py]: Fixed

File: tests/array_interoperability_test.py (hunk 1 above)
Comment (rahulbatra85) [reply tests/array_interoperability_test.py:218]: Added a comment

Comment (rahulbatra85) [reply tests/array_interoperability_test.py:218]: Also regarding a corresponding TF change, I did look at the code that you pointed above, but the testcases would pass with changes just in JAX dlpack.py.

I will investigate further and see if testcases are missing some coverage here

Comment (hawkinsp) [reply tests/array_interoperability_test.py:218]: One thing to note is that the tests will pass if you have only JAX installed, because JAX will agree with itself on what the vendor ID is. Try installing TF and PyTorch with ROCM enabled and run the tests. I assume one or more of those have ROCM support and it is the interoperability between them we must test.

File: tests/array_interoperability_test.py [hunk 3]
```diff
@@ -215,7 +212,7 @@ def testNumpyToJax(self, shape, dtype):
...
+  @jtu.skip_on_devices("gpu") # Numpy doesn't support DLpacks from GPU vendors
```
Comment (hawkinsp) [inline tests/array_interoperability_test.py:215]: A better way to phrase this would be:
"NumPy only accepts cpu DLPacks".

File: tests/array_interoperability_test.py (hunk 1 above)
Comment (rahulbatra85) [reply tests/array_interoperability_test.py:218]: Made some changes on tensorflow to address the device ID for ROCm.
Here is the PR https://github.com/tensorflow/tensorflow/pull/57640

File: tests/array_interoperability_test.py (hunk 3 above)
Comment (rahulbatra85) [reply tests/array_interoperability_test.py]: Done!

**Comment by rahulbatra85** [comment]
FYI: Tensorflow PR that enables DLPack for ROCm was merged this morning
https://github.com/tensorflow/tensorflow/pull/57640

**Comment by rahulbatra85** [comment]
@hawkinsp  Ping!

**Comment by rahulbatra85** [comment]
@hawkinsp  Any feedback on this? Can we merge it?

Also, please note ROCm support for DLPack is in Tensorflow Eager and XLA, so that pre-req is met.

File: jax/_src/dlpack.py [hunk 4]
```diff
@@ -64,6 +64,14 @@ def from_dlpack(dlpack):
     gpu_backend = xla_bridge.get_backend("cuda")
   except RuntimeError:
     gpu_backend = None
+
```
Comment (hawkinsp) [inline jax/_src/dlpack.py:67]: This code will probably need to change in the future. A better design would be to supply both backends to the Python code and it can use whichever it needs. Fine for now.

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/dlpack.py (hunk 4 above)
Comment (rahulbatra85) [reply jax/_src/dlpack.py:67]: @hawkinsp ok, thanks for merging!
